DoSomething/quasar
# Only fetch oauth token if needed

#### What's this PR do?
Stop fetching oauth token with every request. Fetch once when connection is first initialised, and then only re-fetch if we encounter auth errors.

#### Where should the reviewer start?
Look at DSNorthstarScraper.py file


#### How should this be manually tested?
We can use python REPL. Try this code snippet:

```
import quasar
import quasar.DSNorthstarScraper
s = quasar.DSNorthstarScraper.NorthstarScraper('https://northstar.dosomething.org')
r = s.get('/v1/users', {'limit':5})
r.status_code # should be 200
s.auth_headers['Authorization'] = s.auth_headers['Authorization'] + 'blah'
r = s.get('/v1/users', {'limit':5})
r.status_code # should be 200 still
```

#### Any background context you want to provide?
#### What are the relevant tickets?
#### Screenshots (if appropriate)
#### Questions:
